src/module/system/damage/weapon.ts

```diff
@@ -391,14 +391,32 @@ class WeaponDamagePF2e {
             }
         }
 
-        // Roll notes
+        // Critical specialization effects
+        const critSpecEffect = ((): CritSpecEffect => {
+            if (!weapon.isOfType("weapon")) return [];
+
+            // If an alternate critical specialization effect is available, apply it only if there is also a
+            // qualifying non-alternate
+            const critSpecs = actor.synthetics.criticalSpecalizations;
+            const standard = critSpecs.standard.map((cs): CritSpecEffect => cs(weapon, options)).pop();
+            const alternate = critSpecs.alternate.map((cs): CritSpecEffect => cs(weapon, options)).pop();
```

Comment:
Doesn't this process all the critspecs and then just remove the last one from the array and throw away are the ones before it?  It should only be necessary to process the last one.

```
const standard = critSpecs.standard.slice(-1)[0]?.(weapon, options);
```

Reply:
Good catch, but the fix needs to be a bit different.